cmake/system_check.cmake

```diff
@@ -38,7 +38,7 @@ if(CMAKE_CL_64 OR MINGW64)
   endif()
 elseif(MINGW OR (MSVC AND NOT CMAKE_CROSSCOMPILING))
   set(X86 1)
-elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc.*|power.*|Power.*")
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc.*|power.*|Power.*" OR (CMAKE_SYSTEM_NAME MATCHES "Darwin" AND CMAKE_OSX_ARCHITECTURES MATCHES "ppc.*"))
```

Comment:
huh, what does cmake_system_processor return on Darwin/ppc if it is not some variation of ppc or power ? is this undefined on Darwin, or is that change just for the sake of completeness ?

Reply:
`powerpc` on a native system, but if we build for `ppc` on Rosetta, arch detection based on `CMAKE_SYSTEM_PROCESSOR` will not work correctly.